File: server/mc/common2/src/main/java/calebxzhou/rdi/mc/common2/home/HomeResult.java
package calebxzhou.rdi.mc.common2.home;

public record HomeResult(boolean success, String message) {
    public static HomeResult ok(String message) {
        return new HomeResult(true, message);
    }

    public static HomeResult error(String message) {
        return new HomeResult(false, message);
    }
}

File: mc/rcmd/src/main/kotlin/calebxzhou/rdi/mc/rcmd/home/HomeResult.kt
package calebxzhou.rdi.mc.rcmd.home


data class HomeResult(val success: Boolean, val message: String) {
    companion object {
        fun ok(message: String=""): HomeResult {
            return HomeResult(true, message)
        }

        fun error(message: String=""): HomeResult {
            return HomeResult(false, message)
        }
    }
}
